v01/http.go

```diff
@@ -97,8 +97,8 @@ func (e *Event) UnmarshalBinary(req *http.Request) error {
 			continue
 		}
 
-		k = strings.TrimLeft(k, "Ce-")
-		k = strings.TrimLeft(k, "X-")
+		k = strings.TrimPrefix(k, "Ce-")
+		k = strings.TrimPrefix(k, "X-")
 		k = strings.Replace(k, "-", "", -1)
 
 		fieldName, ok := fieldNames[strings.ToLower(k)]
```

Comment:
Sorry for extending beyond this PR, but... why do we setup fieldNames at all. It seems  like a lot of extra work when we could just lower-case the incoming property name and use target.FieldByName() below to find it? If it doesn't find it, it'll return an zero Value and we can just skip it then. This just feels like a lot of extra work with very little value.

Reply:
I did assuming that `FieldByName` was case sensitive. This allows me to map the case insensitive property name extracted from the header to its associated case sensitive version, which I then pass to `FieldByName`. If `FieldByName` is case insensitive then this can be drastically simplified.

Reply:
See: https://golang.org/pkg/reflect/#Value.FieldByNameFunc

Reply:
@tenczar ^

Reply:
I agree that reflection will impact performance. I made the decision to use reflection to avoid a lot of boilerplate in the marshaller methods that checked for each well known property, make any necessary type conversions and set the associated field. The goal was to create marshaling code that would not need to be updated every time new fields were added to the spec.